src/main/java/org/osc/manager/ism/api/IsmSecurityGroupApi.java

```diff
@@ -0,0 +1,194 @@
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.manager.ism.api;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityNotFoundException;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import org.apache.log4j.Logger;
+import org.osc.manager.ism.entities.SecurityGroupEntity;
+import org.osc.sdk.manager.api.ManagerSecurityGroupApi;
+import org.osc.sdk.manager.element.ApplianceManagerConnectorElement;
+import org.osc.sdk.manager.element.ManagerSecurityGroupElement;
+import org.osc.sdk.manager.element.SecurityGroupMemberListElement;
+import org.osc.sdk.manager.element.VirtualSystemElement;
+import org.osgi.service.transaction.control.TransactionControl;
+
+public class IsmSecurityGroupApi implements ManagerSecurityGroupApi {
+
+	private static final Logger LOGGER = Logger.getLogger(IsmSecurityGroupApi.class);
+	private VirtualSystemElement vs;
+	private ApplianceManagerConnectorElement mc;
+
+	private final TransactionControl txControl;
+	private final EntityManager em;
+
+	private static final String SG_NOT_FOUND_MESSAGE = "A security group with id %s was not found.";
+
+	public IsmSecurityGroupApi(VirtualSystemElement vs, ApplianceManagerConnectorElement mc,
+			TransactionControl txControl, EntityManager em) {
+		super();
+		this.vs = vs;
+		this.mc = mc;
+		this.txControl = txControl;
+		this.em = em;
+	}
+
+	public static IsmSecurityGroupApi create(VirtualSystemElement vs, ApplianceManagerConnectorElement mc,
+			TransactionControl txControl, EntityManager em) throws Exception {
+		return new IsmSecurityGroupApi(vs, mc, txControl, em);
+	}
+
+	@Override
+	public String createSecurityGroup(String name, String oscSgId, SecurityGroupMemberListElement memberList)
+			throws Exception {
+		String existingSGId = findSecurityGroupByName(name);
+		if (existingSGId != null) {
+			return existingSGId;
+		}
+
+		return this.txControl.supports(new Callable<String>() {
+
+			@Override
+			public String call() throws Exception {
+				SecurityGroupEntity newSG = new SecurityGroupEntity(name);
+				IsmSecurityGroupApi.this.em.persist(newSG);
+				if (newSG.getSGId() == null) {
```

Comment:
remove this check, the DB should create the id entry or throw an exception automatically.